app/models/form.rb

```diff
@@ -21,17 +21,40 @@ class Form < ApplicationRecord
 
   accepts_nested_attributes_for :questions, allow_destroy: true
 
+  after_destroy :update_surveys
+
   after_commit :index, on: [:create, :update]
   after_commit :delete_index, on: :destroy
 
+  def update_surveys
+    survey_array = surveys.to_a
+    survey_forms.destroy_all
+    survey_array.each(&:update_form_positions)
+  end
+
   def index
-    UpdateIndexJob.perform_later('form', self)
+    UpdateIndexJob.perform_later('form', id)
   end
 
   def delete_index
     DeleteFromIndexJob.perform_later('form', id)
   end
 
+  def update_question_positions
+    FormQuestion.transaction do
+      i = 0
+      form_questions.each do |fq|
+        # Avoiding potential unecessary writes
+        if fq.position != i
+          fq.position = i
```

Comment:
I based my comment on looking through my database. I think the load test data rake task and spreadsheet importer are out of alignment with the rest of the application. So most of the forms I have in my db have FormQuestions that start at 1. If I create a new form in the webapp, then I get FormQuestions that start at 0. So ignore my comment on indexing.